Lib/multiprocessing/shared_memory.py

```diff
@@ -0,0 +1,573 @@
+            )
+            return f"{self.__class__.__name__}({', '.join(formatted_pairs)})"
+
+        #def __getstate__(self):
+        #    if not hasattr(self, "_shm"):
+        #        return existing_type.__getstate__(self)
+        #    state = self._build_state(self)
+        #    return state
+
+        #def __setstate__(self, state):
+        #    self.__init__(**state)
+
+        def __reduce__(self):
+            return (
+                shareable_wrap,
+                (
+                    None,
+                    self._shm.name,
+                    self._proxied_type,
+                    self.shape,
+                    self.strides,
+                    self.dtype.str if hasattr(self, "dtype") else None,
+                    getattr(self, "format", None),
+                ),
+            )
+
+        def copy(self):
+            dupe = existing_type.copy(self)
+            if not hasattr(dupe, "_shm"):
+                dupe = shareable_wrap(dupe)
+            return dupe
+
+        @staticmethod
+        def _build_state(existing_obj, generics_only=False):
+            state = {
+                "shape": existing_obj.shape,
+                "strides": existing_obj.strides,
+            }
+            try:
+                state["dtype"] = existing_obj.dtype
+            except AttributeError:
+                try:
+                    state["format"] = existing_obj.format
+                except AttributeError:
+                    pass
+            if not generics_only:
+                try:
+                    state["shmem_name"] = existing_obj._shm.name
+                    state["cls"] = existing_type
+                except AttributeError:
+                    pass
+            return state
+
+    proxy_type = type(
+        f"{existing_type.__name__}Shareable",
+        CustomShareableProxy.__bases__,
+        dict(CustomShareableProxy.__dict__),
+    )
+
+    if existing_obj is not None:
+        try:
+            proxy_obj = proxy_type(
+                buffer=shm.buf,
+                **proxy_type._build_state(existing_obj)
+            )
+        except Exception:
+            proxy_obj = proxy_type(
+                buffer=shm.buf,
+                **proxy_type._build_state(existing_obj, True)
+            )
+
+        mveo = memoryview(existing_obj)
+        proxy_obj._shm.buf[:mveo.nbytes] = mveo.tobytes()
+
+    else:
+        proxy_obj = proxy_type(buffer=shm.buf, **augmented_kwargs)
+
+    return proxy_obj
+
+
+encoding = "utf8"
+
+class ShareableList:
+    """Pattern for a mutable list-like object shareable via a shared
+    memory block.  It differs from the built-in list type in that these
+    lists can not change their overall length (i.e. no append, insert,
+    etc.)
+
+    Because values are packed into a memoryview as bytes, the struct
+    packing format for any storable value must require no more than 8
+    characters to describe its format."""
```

Comment:
This type has restrictions of the types of objects stored in the list, but those restrictions are not document at the moment.

Reply:
Docs added in GH-11816.